hbase-server/src/main/java/org/apache/hadoop/hbase/master/procedure/SwitchCompactionOffloadProcedure.java

```diff
@@ -0,0 +1,171 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.master.procedure;
+
+import java.io.IOException;
+import org.apache.hadoop.hbase.ServerName;
+import org.apache.hadoop.hbase.procedure2.ProcedureStateSerializer;
+import org.apache.hadoop.hbase.procedure2.ProcedureSuspendedException;
+import org.apache.hadoop.hbase.procedure2.ProcedureUtil;
+import org.apache.hadoop.hbase.procedure2.ProcedureYieldException;
+import org.apache.hadoop.hbase.procedure2.StateMachineProcedure;
+import org.apache.hadoop.hbase.regionserver.CompactionOffloadSwitchStorage;
+import org.apache.hadoop.hbase.util.RetryCounter;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.hadoop.hbase.shaded.protobuf.generated.MasterProcedureProtos.SwitchCompactionOffloadState;
+import org.apache.hadoop.hbase.shaded.protobuf.generated.MasterProcedureProtos.SwitchCompactionOffloadStateData;
+import org.apache.hadoop.hbase.shaded.protobuf.generated.ProcedureProtos;
+
+/**
+ * The procedure to switch compaction offload
+ */
+@InterfaceAudience.Private
+public class SwitchCompactionOffloadProcedure
```

Comment:
The code is almost a clone of SwitchRpcThrottleProcedure, is it possible make a common parent class for these two classes?

Reply:
the switch of rpcThrottle and compactionOffload has same behavior. Therefore Combination is more proper than Inheritance?

Reply:
What do you mean by Combination? What I can see is that there are duplicate code...

Reply:
To eliminate duplicate code. will introduce a procedure `doSwitchProcedure`, It can switch rpcThrottle、compactionOffload and so on, No class inherit this class ?

Reply:
But you need to keep the old SwitchRpcThrottleProcedure here for compatibility? And I'm fine with any approach that could reduce duplication.

Reply:
I think it is no need, It only effect procedure replay at rare situation. And we can skip the procedure via HBCK.
If we want to keep compatibility, We must retain `SwitchRpcThrottleProcedure/SwitchRpcThrottleRemoteProcedure/SwitchRpcThrottleRemoteCallable/SwitchRpcThrottleStateData/SwitchRpcThrottleRemoteStateData` , we need to add too many class for switchCompacionOffload

Reply:
Requiring HBCK operation for updating is not a good idea usually...

But let me see your approach first.